collect_app/src/main/java/org/odk/collect/android/dto/Form.java

```diff
@@ -237,4 +225,20 @@ public String getAutoDelete() {
     public String getLastDetectedFormVersionHash() {
         return lastDetectedFormVersionHash;
     }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
For both `equals` and `hashCode`, I think you can just use `md5Hash`. It feels really odd to have a hash of a hash. And the display name and form ID are implicitly part of the hash, right?

Reply:
Yeah md5Hash is enough. Fixed.